src/avif.c

```diff
@@ -810,7 +797,27 @@ avifBool avifCropRectConvertCleanApertureBox(avifCropRect * cropRect,
     cropRect->y = (uint32_t)(cropY.n / cropY.d);
     cropRect->width = (uint32_t)clapW;
     cropRect->height = (uint32_t)clapH;
-    return avifCropRectIsValid(cropRect, imageW, imageH, yuvFormat, diag);
+    if (!avifCropRectIsValid(cropRect, imageW, imageH, diag)) {
+        return AVIF_FALSE;
+    }
+
+    *upsampleBeforeCropping = ((yuvFormat == AVIF_PIXEL_FORMAT_YUV420 || yuvFormat == AVIF_PIXEL_FORMAT_YUV422) &&
+                               (cropRect->width % 2 || cropRect->x % 2)) ||
+                              (yuvFormat == AVIF_PIXEL_FORMAT_YUV420 && (cropRect->height % 2 || cropRect->y % 2));
+    return AVIF_TRUE;
+}
+
+avifBool avifCropRectConvertCleanApertureBox(avifCropRect * cropRect,
+                                             const avifCleanApertureBox * clap,
+                                             uint32_t imageW,
+                                             uint32_t imageH,
+                                             avifPixelFormat yuvFormat,
+                                             avifDiagnostics * diag)
+{
+    // Keep the same pre-deprecation behavior.
```

Comment:
This does not keep the old behavior exactly. The difference is the first and third conditions at lines 706-710. So to be precise, the comment should say "most of the pre-deprecation behavior".

If we ignore the first and third conditions as I suggested in my comment at line 710, then we keep the old behavior exactly.

Reply:
Done.